frontend/src/components/common/SimpleTable.tsx

```diff
@@ -71,38 +74,58 @@ export default function SimpleTable(props: SimpleTableProps) {
     filterFunction = null,
     emptyMessage = null,
     errorMessage = null,
+    tableName = null
   } = props;
   const [page, setPage] = React.useState(0);
   const [currentData, setCurrentData] = React.useState(data);
   const rowsPerPageOptions = props.rowsPerPage || [5, 10, 50];
   const [rowsPerPage, setRowsPerPage] = React.useState(rowsPerPageOptions[0]);
   const classes = useTableStyle();
+  const history = useHistory();
+  const location = useLocation();
+  const queryParamDefaultObj: {[key: string]: string} = { 'page': '0', 'rowsPerPage': rowsPerPageOptions[0].toString()};
 
   function handleChangePage(_event: any, newPage: number) {
+    addQuery({'page': newPage.toString(), 'rowsPerPage': rowsPerPage.toString()}, queryParamDefaultObj, history,
+    location, tableName || '');
     setPage(newPage);
   }
 
   function handleChangeRowsPerPage(event: any) {
+    addQuery({'page': '0', 'rowsPerPage': event.target.value.toString()}, queryParamDefaultObj, history,
+    location, tableName || '');
     setRowsPerPage(+event.target.value);
     setPage(0);
   }
 
   React.useEffect(() => {
-    if (currentData === data) {
-      if (page !== 0) {
-        setPage(0);
-      }
-      return;
-    }
-
     // If the currentData is not up to date and we are in the first page, then update
     // it directly. Otherwise it will require user's intervention.
     if ((!currentData || currentData.length === 0) || page === 0) {
       setCurrentData(data);
     }
   },
   // eslint-disable-next-line
-  [data, currentData]);
+  [data, currentData])
+
+  function refreshClickHandler(data : SimpleTableProps['data']){
+    addQuery({'page': '0', 'rowsPerPage': rowsPerPage.toString()}, queryParamDefaultObj,
+    history, location, tableName as string);
+    setCurrentData(data);
+  }
+
+  React.useEffect(() => {
+    const tableNameUrlParam = getFilterValueByNameFromURL('tableName', history) || null;
+    if (!tableNameUrlParam || tableNameUrlParam[0] === tableName) {
+      const rowsPerPageUrlParam = getFilterValueByNameFromURL('rowsPerPage', location);
+      const pageUrlParam = getFilterValueByNameFromURL('page', location);
+      rowsPerPageUrlParam ? setRowsPerPage(parseInt(rowsPerPageUrlParam[0])) :
+        setRowsPerPage(rowsPerPageOptions[0]);
+      pageUrlParam ? setPage(parseInt(pageUrlParam[0])) : 0;
```

Comment:
Seems like the ternary operation should be inside `setPage`? (otherwise the `0` is not used)
Maybe it should also only set the page/rows if the values change (otherwise changing the row also changes the page, and vice-versa, getting us extra refreshes that don't need to be done).